src/Lightning.ts

```diff
@@ -712,8 +706,7 @@ export const LightningMixin = (superclass) => class extends superclass {
 
       } catch (err) {
         const error = `issue updating invoice`
-        this.logger.error({ err, invoice }, error)
-        throw new LoggedError(error)
+        throw new DbError(error, {forwardToClient: true, logger: this.logger, level: 'error', err, invoice})
```

Comment:
I think we should have 2 errors here. one for the first mongodb fetch regarding `InvoiceUser` and the second for `addTransactionLndReceipt`. so that if an error occurs we know more precisely at what level it occured.

Reply:
actually the way it's structured, I also think this error could cover the case if we can't acquire the lock? not sure this try catch is actually helpful. something to give a try probably, or least add a TODO / FIXME to revisit later.

Reply:
I have moved the try catches inside the redlock function. Do we want to enclose the entire redlock function in another try catch?